Sources/NIOSSL/TLSConfiguration.swift

```diff
@@ -200,7 +233,45 @@ public struct TLSConfiguration {
 
     /// The pre-TLS1.3 cipher suites supported by this handler. This uses the OpenSSL cipher string format.
     /// TLS 1.3 cipher suites cannot be configured.
-    public var cipherSuites: String
+    public var cipherSuites: String = defaultCipherSuites
+    
+    /// Public property used to set the internal cipherSuiteFormattedString from NIOTLSCipher.
+    public var cipherSuite: [NIOTLSCipher] {
+        get {
+            // Set a temporary context with the default ciphers from cipherSuites
+            guard let tempContext = CNIOBoringSSL_SSL_CTX_new(CNIOBoringSSL_TLS_method()) else {
+                fatalError("Failed to create new BoringSSL context")
+            }
+            // Set a temporary context to build a set of default ciphers on based upon defaultCipherSuites
+            CNIOBoringSSL_SSL_CTX_set_cipher_list(tempContext, cipherSuites)
+            let bsslCipherStack = CNIOBoringSSL_SSL_CTX_get_ciphers(tempContext)
+            
+            var defaultCiphers: [NIOTLSCipher] = []
+            let cipherStackSize = CNIOBoringSSL_sk_SSL_CIPHER_num(bsslCipherStack)
+            for idx in 0..<cipherStackSize {
+                guard let stackCipherPtr = CNIOBoringSSL_sk_SSL_CIPHER_value(bsslCipherStack, idx) else {
+                    preconditionFailure("Unable to get cipher \(idx) from stack \(String(describing: bsslCipherStack))")
+                }
+                // Create a new cipher based on the NIOTLSCipher type
+                let id = CNIOBoringSSL_SSL_CIPHER_get_protocol_id(stackCipherPtr) as UInt16
+                defaultCiphers.append(NIOTLSCipher(id))
+            }
+            return defaultCiphers
+        }
+        set {
+            // Iterate through an array of provided ciphers and build out a dynamic cipherSuites
+            // string based on the passed in suites.
+            var ciphers = ""
+            for cipher in newValue {
+                let cipherValue: UInt16 = cipher.rawValue
+                let boringSSLCipher = CNIOBoringSSL_SSL_get_cipher_by_value(cipherValue)
+                let standardName = String(cString: CNIOBoringSSL_SSL_CIPHER_standard_name(boringSSLCipher))
+                ciphers += "\(standardName):"
```

Comment:
Instead of repeatedly appending to this string, we should use `map` to create an array of `Strings` and then use `joined` to join them together.

Reply:
This is addressed in https://github.com/apple/swift-nio-ssl/pull/290/commits/1811292f3d5e17db5b3af27cc90f2009870e91d0